### DISCUSSION TITLE
How to enable Swagger API Key authorization in UI

### DISCUSSION THREAD

**Discussion by gregggilbert-ivadolabs** [aws-powertools/powertools-lambda-python#3609] [Q&A] [answered]
Adding API Key authorization in a Swagger template is pretty straightforward. One simple need to add to the template:

```
components:
  securitySchemes:
    ApiKeyAuth:        # arbitrary name for the security scheme
      type: apiKey
      in: header       # can be "header", "query" or "cookie"
      name: X-API-KEY  # name of the header, query parameter or cookie
# 2) Apply the API key globally to all operations
security:
  - ApiKeyAuth: []     # use the same name as under securitySchemes
```

How can this be achieved with PowerTools' automatic Swagger documentation generation? 
`app.enable_swagger()` doesn't offer any way to add content to the Swagger template.

**Comment by rubenfonseca** [comment, marked as answer]
Hi @gregggilbert-ivadolabs security schemes, headers, cookies and files are currently not implemented in Powertools' OpenAPI implementation. @leandrodamascena should we convert this to a feature request?

**Reply by gregggilbert-ivadolabs** [reply]
This is a good idea because, otherwise, we have to override APIGatewayRestResolver's `enable_swagger` and `get_openapi_schema` methods to make it work.

**Reply by leandrodamascena** [reply]
Hello @rubenfonseca and @gregggilbert-ivadolabs! I'm converting this into an issue so we can implement it.
